src/io.c

```diff
@@ -1,4478 +1,4641 @@
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = SphP[pindex].DtInternalEnergy;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_DELAYTIME:
+#ifdef GALSF_SUBGRID_WINDS
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = SphP[pindex].DelayTime;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_TSTP:		/* timestep  */
+#ifdef OUTPUT_TIMESTEP
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = GET_PARTICLE_TIMESTEP_IN_PHYSICAL(pindex);
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_BFLD:		/* magnetic field  */
+#ifdef MAGNETIC
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    for(k=0;k<3;k++) {fp[k] = (MyOutputFloat) (Get_Gas_BField(pindex,k) * All.cf_a2inv * gizmo2gauss);}
+                    fp += 3;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_VDIV:		/* Divergence of Vel */
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = P[pindex].Particle_DivVel;
+                    n++;
+                }
+            break;
+
+        case IO_VORT:		/* Vorticity */
+#if defined(TURB_DRIVING) || defined(OUTPUT_VORTICITY)
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    for(k=0;k<3;k++) {fp[k] = SphP[pindex].Vorticity[k];}
+                    fp += 3;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_SLUG_STATE_RNG:
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type) /* Since only npart > 0 is saved, this line is not important */
+                {
+                    *ip_int++ = (rng_state_t) P[pindex].rngStateAtBirth;
+                    n++;
+                }
+            break;
+
+        case IO_SLUG_STATE_INT:
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type) /* Since only npart > 0 is saved, this line is not important */
+                {
+                    *ip_int++ = (uint64_t) P[pindex].id;
+                    *ip_int++ = (uint64_t) P[pindex].stoch_sn;
+                    n++;
+                }
+            break;
+
+        case IO_SLUG_STATE_DOUBLE: /*I did not include the last three quantities since I dont know how to deal with N */
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type) /* Since only npart > 0 is saved, this line is not important */
+                {
+                    *fp++ = (double) P[pindex].targetMass;

```

Comment:
Change to P[pindex].slug_state.targetMass etc.

Reply:
To be consistent with the rest of GIZMO, you may want to cast this to `MyFloat` instead of `double`. See this: https://github.com/BenWibking/gizmo-fork/blob/7e196c64c0870e621f38db50621f42b610891cfd/src/io.c#L3952